src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -69,97 +234,135 @@ public void runCargo(boolean a,boolean reversed) {
 
   public boolean wheelReady(){
     double flywheelSpeed = KShooterEncoder.getVelocity();
+    currentRPM = DShooterRPMInput.getDouble(0);
     if (flywheelSpeed > currentRPM - 15 && flywheelSpeed < currentRPM + 15) {
+      
       return true;
     }
+    else{
     return false;
+    }
   }
-
-  public void setAimMode(int m) {
-    aimMode = m;
+  public void setAimMode(Double m) {
+    DShootingMode.setDouble(m);
   }
 
   public void cycleAimModeUp() {
-    aimMode++;
-    if (aimMode > 3) {
-      aimMode = 0;
+    double ShooterAimMode = DShootingMode.getDouble(0);
+    ShooterAimMode ++;
+    if (ShooterAimMode>3){
+      ShooterAimMode = 0;
     }
+    setAimMode(ShooterAimMode);
+    
   }
 
   public void cycleAimModeDown() {
-    aimMode--;
-    if (aimMode < 0) {
-      aimMode = 3;
+    double ShooterAimMode = DShootingMode.getDouble(0);
+    ShooterAimMode--;
+    if (ShooterAimMode < 0) {
+      ShooterAimMode = 3;
     }
+    setAimMode(ShooterAimMode);
   }
 
   public double getTY() {
     // Gets TY, the vertical angle of the target from the limelight
     return NetworkTableInstance.getDefault().getTable("limelight").getEntry("ty").getDouble(0);
   }
+  public double getFlyWheelSpeed(){
+    return shooter_motorController.getEncoder().getVelocity();
+  }
 
   public double getDistance() {
     // Uses Limelight to find distance to High Goal
     SmartDashboard.putNumber("ty", getTY());
-    return (Constants.highHeight - Constants.LLHeight) / Math.tan(Math.toRadians((getTY() + Constants.LLAngle))); // Return distance in
+    return (Constants.Shooter.highHeight - Constants.Shooter.LLHeight) / Math.tan(Math.toRadians((getTY() + Constants.Shooter.LLAngle))); // Return distance in
                                                                                                    // feet
   }
 
+
+  ///Aimodes
+  public void automode(){
+    adjustHood(ProjectilePrediction(Constants.Shooter.shooterHeight, 0, Constants.Shooter.highHeight, getDistance(),Constants.Shooter.gravity, Constants.Shooter.airboneTime)[1]);
+    windFlywheel((int) (Math.ceil(ProjectilePrediction(Constants.Shooter.shooterHeight, 0, Constants.Shooter.highHeight,getDistance(), 32, Constants.Shooter.airboneTime)[0])));
+  }
+
+
   public double[] ProjectilePrediction(double y0, double x0, double y, double x, double g, double t) {
     x = x + 1; // Applies an offset to target goal center
     double hoodAngle = Math.toDegrees(Math.atan((y - y0 + 1 / 2 * g * (Math.pow(t, 2))) / x)); //Finds angle to shoot at
     double velocity = Math.abs(x / (Math.cos(Math.toRadians(hoodAngle)) * t)); //Finds velocity in feet per second to shoot at
     double[] returnArray = new double[2];
-    returnArray[0] = UnitConversion(velocity, Constants.gearDiameter); //Converts FPS to RPM
+    returnArray[0] = UnitConversion(velocity, Constants.Shooter.gearDiameter); //Converts FPS to RPM
```

Comment:
lookup isn't used anywhere.  Need to update projectile prediction to use lookup.

When that is attempted, should consider simplifying the input parameters of ProjectilePrediction.  No need to pass in "g" for example.

Reply:
Talked about this with Jim, addressing in a next pull request.